03week/ticTacToe.js

```diff
@@ -24,39 +24,103 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  if (board[0][0]==board[0][1] && board[0][1]==board[0][2] && board[0][0]!=(' ')){
+    return true;
+  }
+  if (board[1][0]==board[1][1] && board[1][1]==board[1][2] && board[1][0]!=(' ')){
+    return true;
+  }
+  if (board[2][0]==board[2][1] && board[2][1]==board[2][2] && board[2][0]!=(' ')){
+    return true;
+  }
 }
 
 function verticalWin() {
-  // Your code here
+  if (board[0][0]==board[1][0] && board[1][0]==board[2][0] && board[0][0]!=(' ')){
+    return true;
+  }
+  if (board[0][1]==board[1][1] && board[1][1]==board[2][1] && board[0][1]!=(' ')){
+    return true;
+  }
+  if (board[0][2]==board[1][2] && board[1][2]==board[2][2] && board[0][2]!=(' ')){
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if (board[0][0]==board[1][1] && board[1][1]==board[2][2] && board[0][0]!=(' ')){
+    return true;
+  }
+  if (board[0][2]==board[1][1] && board[1][1]==board[2][0] && board[0][2]!=(' ')){
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin() === true){
+    return true;
+  }
+  if (verticalWin() === true){
+    return true;
+  }
+  if (diagonalWin() === true){
+    return true;
+  }
 }
 
+
 function ticTacToe(row, column) {
-  // Your code here
+let rowGood = row.trim();
```

Comment:
The tests failed because they passed in numbers instead of strings, and trim does not work on numbers.
(not taking points off, just pointing it out.